lab_5_scrapper/scrapper.py

```diff
@@ -127,11 +237,28 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        links = article_bs.find_all('a', class_='listing-preview__content')
+        url = ''
+        for link in links:
+            url = link['href']
+        return url
 
     def find_articles(self) -> None:
         """
         Find articles.
         """
+        urls = []
+        while len(urls) < self.config.get_num_articles():
+            for url in self.get_search_urls():
+                response = make_request(url, self.config)
+
+                if not response.ok:
+                    continue
+
+                soup = BeautifulSoup(response.text, 'lxml')
+                urls.append(self._extract_url(soup))
+
+        self.urls.extend(urls)
```

Comment:
why do not we modify `self.urls` directly? why do we need to update it only after all links are collected?

Reply:
done